### ISSUE TITLE
getting an extra blank line when iterating over a slice

### ISSUE THREAD

**Issue by craigschmidt** [cessen/ropey#15]
I'm new to rust,  so sorry if I'm doing something dumb, but I'm seeing a weird bug.  When iterating over a slice with `lines`, I get an extra empty line.  If you make a new program with cargo like this, it fails.  However, I added the same test to ropey as a test in the iter.rs file, and it passed.  I don't understand how that can be.  

Thanks,
Craig

```rust
extern crate ropey;

use ropey::Rope;

fn main() {

    let text = "Hello there!\nHow goes it?";
    let rope = Rope::from_str(text);

    // should be the same as rope
    let slice = rope.slice(..);

    // but we get an extra empty line when iterating over the slice
    assert!(slice.lines().count() == rope.lines().count());
}


#[cfg(test)]
mod tests {
    use Rope;

    #[test]
    fn lines_from_slice() {
        
        let text = "Hello there!\nHow goes it?";
        let rope = Rope::from_str(text);

        // should be the same as rope
        let slice = rope.slice(..);

        // but we get an extra empty line when iterating over the slice
        assert!(slice.lines().count() == rope.lines().count());
    }

}
```

**Comment by craigschmidt** [comment]
My `Cargo.toml` looks like this:

```toml
[package]
name = "iterbug"
version = "0.1.0"
authors = ["Craig Schmidt <craig@craigschmidt.com>"]

[dependencies]
ropey = { git = "https://github.com/cessen/ropey", branch = "master" }
```

**Comment by cessen** [comment]
Thanks so much for the bug report!  Looking into it now. :-)

**Comment by cessen** [comment]
Found the bug!  Working on a fix now.

Thank you again!  I really appreciate you taking the time to file such a thorough bug report.

**Comment by cessen** [comment]
Fixed in bc7f3a2a59c0587951861773bf71115eb9805403.
